app/src/main/java/ru/yandex/practicum/contacts/presentation/filter/model/FilterContactTypeUi.java

```diff
@@ -20,6 +22,12 @@ public boolean isSelected() {
         return selected;
     }
 
+    @Override
+    public boolean theSameAs(FilterContactTypeUi other){
+        if (other == null) return false;
+        return this.contactType.equals(other.contactType);
```

Comment:
Проверка на `null` валидна, её можно оставить, но сам `FilterContactType` является енамом, поэтому лучше сравнить просто через `==`, а не через вызов `equals`:
```suggestion
        if (other == null) return false;
        return this.contactType == other.contactType;
```